SDWebImage/SDWebImageManager.m

```diff
@@ -380,6 +367,33 @@ - (void)callCompletionBlockForOperation:(nullable SDWebImageCombinedOperation*)o
     });
 }
 
+- (BOOL)shouldBlockFailedURLWithURL:(nonnull NSURL *)url
+                              error:(nonnull NSError *)error {
+    // Check whether we should block failed url
+    BOOL shouldBlockFailedURL;
+    if ([self.delegate respondsToSelector:@selector(imageManager:shouldBlockFailedURL:withError:)]) {
+        shouldBlockFailedURL = [self.delegate imageManager:self shouldBlockFailedURL:url withError:error];
+    } else {
+        NSString *domain = error.domain;
+        NSInteger code = error.code;
+        if ([domain isEqualToString:NSURLErrorDomain]) {
+            shouldBlockFailedURL = (   code != NSURLErrorNotConnectedToInternet
+                                    && code != NSURLErrorCancelled
+                                    && code != NSURLErrorTimedOut
+                                    && code != NSURLErrorInternationalRoamingOff
+                                    && code != NSURLErrorDataNotAllowed
+                                    && code != NSURLErrorCannotFindHost
+                                    && code != NSURLErrorCannotConnectToHost
+                                    && code != NSURLErrorNetworkConnectionLost);
+        } else {
+            // Custom loader, don't do extra error code check
+            shouldBlockFailedURL = NO;
```

Comment:
Attention! please don't do this things in a refactor `PR`. This break the processing logical. I'll revert this.

Reply:
I'll create another PR to revert this again. This is a designed logic and should be kept.

Reply:
@dreampiggy See #2580 , I don't merge it directly because I think every `PR` needs review.

And for this `PR`, first, this change break the original processing logical, other words, this `PR` is not only do refactor. And aside this, I don't think we should do this check.